### ISSUE TITLE
Possibility to specify an encoding

### ISSUE THREAD

**Issue by skuzzle** [migzai/irc-api#30]
Hi,

I was trying to use this IRC lib because it seems to be the only one which is accessible through maven central and also seemed to be well designed. Now I stumbled on a problem: The default SocketChannelConnection always uses the system's default encoding. As far as I have researched the framework offers no easy way of exchanging the connection implementation without completely reimplementing the IRCApi interface.

**Comment by migzai** [comment]
You can change the default encoding anytime at run-time. Use System.setProperty("file.encoding", "your_desired_charset");

**Comment by skuzzle** [comment]
That's true, but maybe changing the default encoding is too invasive for the remaining parts of my application (besides that, it's bad practice relying on the default encoding because this makes the application highly platform dependent, and, in this case, IRC server dependent)

**Comment by anormal81** [comment]
System.setProperty("file.encoding", "your_desired_charset"); ist not a solution for us. Some of our servers in european region operate with an iso8859-1 code page which is required. Sending messages with umlauts to an UTF-8 IRC server results in characters that can not be displayed. Each IRC client I've seen so far allows the specification of the encoding. Please add this as setting in IServerParamters.

**Comment by migzai** [comment]
Alright will work on that.
